src/rojo_ref.rs

```diff
@@ -0,0 +1,40 @@
+use std::{fmt, sync::Arc};
+
+use serde::{Deserialize, Serialize};
+
+pub const REF_ID_ATTRIBUTE_NAME: &str = "Rojo_Id";
+pub const REF_POINTER_ATTRIBUTE_PREFIX: &str = "Rojo_Target_";
+
+// TODO add an internment strategy for RojoRefs
+// Something like what rbx-dom does for SharedStrings probably works
+
+#[derive(Debug, Default, PartialEq, Hash, Clone, Serialize, Deserialize, Eq)]
+pub struct RojoRef(Arc<Vec<u8>>);
```

Comment:
The primary reason was that user-defined attributes are BinaryStrings. The idea being that `String::from_utf8` is fallible and patch application isn't. In hindsight though, I don't know why I did it that way and it makes more sense to raise a warning or error. I think an error is probably better, so I'll revert the change and just try to convert user attributes into strings.

Do you think it's worth raising an actual error, which involves making `apply_patch_set` return a `Result` and handling that (I haven't looked at how much work that'd be, it might be zero) or should I just log it as an error?